File: Data Structures & Algorithms/climbing-stairs/submission-0.js
class Solution {
    /**
     * @param {number} n
     * @return {number}
     */
    climbStairs(n) { 
        return this.dfs(0, n)
    } 

    dfs(cur, n) {
         if (cur === n) return 1 
         if (cur > n) return 0

        return this.dfs(cur + 1, n) + this.dfs(cur + 2, n)
    }
}

File: Data Structures & Algorithms/count-paths/submission-0.py
class Solution:
    def uniquePaths(self, m: int, n: int) -> int:
         
        def dfs(m, n):  
            if m == 0 or n == 0: 
                return 0 

            if m == 1 and n == 1: 
                return 1 

            return dfs(m - 1, n) + dfs(m, n - 1) 

        return dfs(m, n)